### DISCUSSION TITLE
Raspberry PI 4 - armv7l - building issues

### DISCUSSION THREAD

**Discussion by Patrik-ai** [open-quantum-safe/liboqs#967] [Q&A]
Hi everybody.
I have problems with building liboqs library on my Raspberry Pi 4 with armv 7l processor (i suppose it is 32-bit arch.). I am trying to build library with 
`cmake -GNinja -DCMAKE_TOOLCHAIN_FILE=../.CMake/toolchain_armhf.cmake -DOQS_USE_OPENSSL=OFF ..`
I get error with:
.CMake/detect_gcc_clang_intrinsics.c returned exit code:
  PLEASE_FILL_OUT-FAILED_TO_RUN
Can you please help me with this ?
Thank you very much.

**Comment by jschanck** [comment]
Some liboqs configurations need an emulator during cross-compilation. The toolchain file you're using looks for `qemu-arm-static` as the default. If you're on Ubuntu or Debian `qemu-arm-static` is provided by the qemu-user-static package. 

You can use a different emulator by editing the toolchain file:
```cmake
set(CMAKE_CROSSCOMPILING_EMULATOR "qemu-arm-static;-L;/usr/arm-linux-gnueabihf/")
```

Alternatively, you can compile a configuration that doesn't need an emulator. Just pass `-DOQS_DIST_BUILD=ON` to CMake.

**Reply by Patrik-ai** [reply]
Hi there, 
thank you for help it works perferctly for me. And for openssl as well

In the next step i was trying to build nginx image like this:
`sudo docker build --build-arg LIBOQS_BUILD_DEFINES=DOQS_DIST_BUILD=ON  -t oqs-nginx-img .`

Error occurs:
![image](https://user-images.githubusercontent.com/79135565/115218811-3884c380-a107-11eb-95fc-eca7323fb3dc.png)

Is any solution for this ? 
Thank you

**Reply by baentsch** [reply]
This looks like what it says: A temporary error reaching some alpine SW distro server. Solution: Wait a bit :-) Or re-write the Dockerfile to use another OS distribution (ubuntu almost certainly works).

**Reply by baentsch** [reply]
Just checking back: Did this resolve (automagically) by now? Can we close the discussion/mark it answered?
